### PR TITLE
[WIP] Add schema_extra metadata to Pydantic schemas

### LINKED ISSUE #102: Add `schema_extra` metadata to Pydantic schemas
FastAPI allows us to specify example data attributes to be picked up and auto documented by Swagger. We should use this in our externally facing Pydantic schemas. https://fastapi.tiangolo.com/tutorial/schema-extra-example/

### PR DESCRIPTION
**Pull request by wiseaidev**
# Purpose

add schema examples.

# Changes

Add schema examples for: 

- [ ] schemas
  - [x] connection_configuration
    - [X] connection_config.py
    - [X] connection_secrets.py
    - [X] connection_secrets_bigquery.py
    - [X] connection_secrets_email.py  
    - [X] connection_secrets_mariadb.py 
    - [X] connection_secrets_mongodb.py 
    - [X] connection_secrets_mssql.py 
    - [X] connection_secrets_mysql.py
    - [X] connection_secrets_postgres.py
    - [X] connection_secrets_redshift.py
    - [X] connection_secrets_snowflake.py
    - [X] connections_secrets_https.py
  - [ ] email
    - [ ] email.py

# Checklist
- [ ] Update [`CHANGELOG.md`](https://github.com/ethyca/fidesops/blob/main/CHANGELOG.md) file
  - [ ] Merge in main so the most recent `CHANGELOG.md` file is being appended to
  - [ ] Add description within the `Unreleased` section in an appropriate category. Add a new category from the list at the top of the file if the needed one isn't already there.
  - [ ] Add a link to this PR at the end of the description with the PR number as the text. example: [#1](https://github.com/ethyca/fidesops/pull/1)
- [ ] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md).
- If docs updated (select one):
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
- [ ] Good unit test/integration test coverage
- [ ] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

# Ticket

Fixes #102

### CODE REVIEW

File: src/fidesops/ops/schemas/connection_configuration/connection_config.py [hunk 1]
```diff
@@ -32,6 +33,16 @@ class Config:
         orm_mode = True
         use_enum_values = True
         extra = Extra.forbid
+        schema_extra = {
+            "example": {
+                "name": "Foo",
+                "key": FidesOpsKey("eb65d06059de1c30aa2b973a33075"),
```
Comment (seanpreston) [inline src/fidesops/ops/schemas/connection_configuration/connection_config.py:39]: Hey @wiseaidev, thanks for the contributions!

These values will end up in our autogenerated OpenAPI documentation, so we prefer they're kept human readable. e.g.  `connection_config_key` would be more appropriate here.

File: src/fidesops/ops/schemas/connection_configuration/connection_config.py [hunk 2]
```diff
@@ -112,7 +154,28 @@ class SaasConnectionTemplateValues(BaseModel):
...
     instance_key: FidesOpsKey  # For DatasetConfig.fides_key
 
+    class Config:
+        """Add examples for each field"""
+        schema_extra = {
+            "example": {
+                "name": "Foo",
+                "key": FidesOpsKey("eb65d06059de1c30aa2b973a33075"),
+                "description": "A Schema with values to create both a Saas ConnectionConfig and DatasetConfig from a template.",
+                "secrets": PostgreSQLSchema(username="Mahmoud", password="S3c11R3P@ssW0rd", _required_components=["host", ]),
```
Comment (seanpreston) [inline src/fidesops/ops/schemas/connection_configuration/connection_config.py:164]: Here also, something like:

`username="<your_username>", password="<your_password>"` would be better.

File: src/fidesops/ops/schemas/connection_configuration/connection_secrets.py [hunk 3]
```diff
@@ -45,10 +45,24 @@ class Config:
 
         extra = Extra.forbid
         orm_mode = True
+        schema_extra = {
+            "example": {
+                "url": "http://wiseai.dev/"
```
Comment (seanpreston) [inline src/fidesops/ops/schemas/connection_configuration/connection_secrets.py:50]: Please avoid using real world URLs in example data, preference `https://example.com`

File: src/fidesops/ops/schemas/connection_configuration/connection_secrets_bigquery.py [hunk 4]
```diff
@@ -22,6 +22,23 @@ class KeyfileCreds(BaseModel):
...
+                "type": "service_account",
+                "project_id": "XXXXXXXXXXXXXXXXXXXX",
+                "private_key_id": "key-id",
+                "private_key": "-----BEGIN PRIVATE KEY-----\nprivate-key\n-----END PRIVATE KEY-----\n",
+                "client_email": "service-account-email",
+                "client_id": "XXXXXXXXXXXXXXXXXXXX",
+                "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+                "token_uri": "https://accounts.google.com/o/oauth2/token",
+                "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+                "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/service-account-email"
```
Comment (seanpreston) [inline src/fidesops/ops/schemas/connection_configuration/connection_secrets_bigquery.py:38]: @pattisdr what should our convention be wrt real world URLs for connection secrets? I'd prefer we linked to the provider's docs at most, is that reasonable?

Comment (pattisdr) [reply src/fidesops/ops/schemas/connection_configuration/connection_secrets_bigquery.py:38]: Makes sense to me

File: src/fidesops/ops/schemas/connection_configuration/connection_secrets_mariadb.py [hunk 5]
```diff
@@ -19,6 +19,18 @@ class MariaDBSchema(ConnectionConfigSecretsSchema):
 
     _required_components: List[str] = ["host"]
 
+    class Config:
+        """Add example for each field"""
+        schema_extra = {
+            "example": {
+                "username": "Mahmoud",
```
Comment (pattisdr) [inline src/fidesops/ops/schemas/connection_configuration/connection_secrets_mariadb.py:26]: Similar, more generic "<your_username>"

File: src/fidesops/ops/schemas/connection_configuration/connection_secrets_mongodb.py [hunk 6]
```diff
@@ -17,6 +17,18 @@ class MongoDBSchema(ConnectionConfigSecretsSchema):
 
     _required_components: List[str] = ["host"]
 
+    class Config:
+        """Add example for each field"""
+        schema_extra = {
+            "example": {
+                "username": "Mahmoud",
```
Comment (pattisdr) [inline src/fidesops/ops/schemas/connection_configuration/connection_secrets_mongodb.py:24]: Maybe <your_username>

**Comment by wiseaidev** [comment]
Hey @seanpreston and @pattisdr, thanks for the reviews. Alright, I am going to adjust the values accordingly and continue my series of documentation on all schemas.  One thing, would you mind adding `hacktoberfest` label for the repo?

**Comment by seanpreston** [comment]
> One thing, would you mind adding hacktoberfest label for the repo?

Will do this today.

@wiseaidev did you see my email wrt the new repository? I sent it to the address listed publicly on your Github profile.

**Comment by wiseaidev** [comment]
Hey @seanpreston, email received. I will move my contribs over there.

**Comment by seanpreston** [comment]
Appreciate it, thanks

**Pull request closed without merging**